etcd_registry.go

```diff
@@ -46,17 +46,21 @@ type registerMeta struct {
 }
 
 // NewEtcdRegistry creates a etcd based registry.
-func NewEtcdRegistry(endpoints []string) (registry.Registry, error) {
-	return NewEtcdRegistryWithAuth(endpoints, "", "")
+func NewEtcdRegistry(endpoints []string, opts ...Option) (registry.Registry, error) {
+	return NewEtcdRegistryWithAuth(endpoints, "", "", opts...)
 }
 
 // NewEtcdRegistryWithAuth creates a etcd based registry with given username and password.
-func NewEtcdRegistryWithAuth(endpoints []string, username, password string) (registry.Registry, error) {
-	etcdClient, err := clientv3.New(clientv3.Config{
+func NewEtcdRegistryWithAuth(endpoints []string, username, password string, opts ...Option) (registry.Registry, error) {
```

Comment:
How about moving the implementation to `NewEtcdRegistry` and call it with username and password option ?  And we can mark this function as deprecated in comment. 

Since the username and password in parameter would be override by option，a little bit confusing. 

Same as resolver. What do you think ?

Reply:
Yep, I think it would be better.  Just mark this function is deprecated and clear out using options instead in comment.
https://stackoverflow.com/questions/7849663/how-do-you-mark-code-as-deprecated-in-go maybe help.